apps/site/components/Common/Partners/index.tsx

```diff
@@ -0,0 +1,66 @@
+'use server';
+
+import Tooltip from '@node-core/ui-components/Common/Tooltip';
+import * as PartnerLogos from '@node-core/ui-components/Icons/PartnerLogos';
+
+import providePartners from '#site/next-data/providers/partners';
+
+import type { PartnerCategory } from '#site/types';
+import type { FC } from 'react';
+
+import PartnerButton from './PartnerButton';
+
+import style from './index.module.css';
+
+type PartnersListProps = {
+  size?: 'big' | 'small';
+  category?: PartnerCategory;
+  length: number;
+};
+
+const PartnersList: FC<PartnersListProps> = async ({
+  size = 'small',
+  category,
+  length,
+}) => {
+  let partners = await providePartners();
+
+  if (category) {
```

Comment:
I know it is an async function so settings things on root of component doesn't matter, bujt can we wrap it on a function maybe? (outside)